test-data/unit/check-unreachable-code.test

```diff
@@ -932,10 +932,86 @@ if False:
 
 if False:
     # Ignore obvious type errors
-    assert_never(expect_str(x))
+    assert_never(expect_str(x))  # E: Statement is unreachable
     reveal_type(x)
 [builtins fixtures/exception.pyi]
 
+[case testUnreachableFlagOkWithDeadStatements2]
+# flags: --warn-unreachable
+from typing import NoReturn
+def assert_never(x: NoReturn) -> NoReturn:
+    assert False
+
+def nonthrowing_assert_never(x: NoReturn) -> None: ...
+
+def expect_str(x: str) -> str: pass
+
+x: int
+if False:
+    assert False
+
+if False:
+    raise Exception()
+
+if False:
+    assert_never(x)
+
+if False:
+    nonthrowing_assert_never(x)  # E: Statement is unreachable
+
+if False:
+    # Ignore obvious type errors
+    assert_never(expect_str(x))
+[builtins fixtures/exception.pyi]
+
+
+[case testSeveralUnreachableStatements]
+# flags: --warn-unreachable
+from typing import NoReturn
+def assert_never() -> NoReturn:
```

Comment:
The name seems a bit confusing. This would have to take an argument with type `NoReturn` to assert that code path is never reached, I think. Maybe rename to to `never_return`?